docs/contributing/999-rfcs/2024-03-31-wing-secrets-cli.md

```diff
@@ -0,0 +1,116 @@
+---
+title: "#6105 Wing Secrets CLI"
+description: Creating Wing secrets from the CLI
+---
+
+# Wing Secrets CLI
+- **Author(s)**: @hasanaburayyan
+- **Submission Date**: 2024-03-31
+- **Stage**: Draft
+
+Creating secrets through the Wing CLI.
+
+## Background
+
+Wing applications often require secrets to be retrieved during runtime. These secrets are stored in platform specific secret stores, such as AWS Secrets Manager for `tf-aws` or a local `.env` file for the `sim` platform.
+
+Secrets must be configured before the application is run, and now the Wing CLI along with Wing platforms make it easy to configure secrets.
+
+### Out of Scope
+
+In this RFC a few things are out of scope:
+- Checking if the secrets exist in the platform's secret store when running `wing compile`
+- Reading secret values, for now we will only focus on creating secrets
+
+## Platform Hook
+
+Since secrets creation is platform specific, platforms can now implement a new hook `configureSecrets(secrets: { [key: string]: string }): string` which will be called by the Wing CLI to configure the secrets.
+
+For example the `sim` platform implementation which needs to store secrets in a `.env` file, would look something like this:
+
+```js
+public async createSecrets(secrets: { [key: string]: string }): Promise<string> {
```

Comment:
I'd call this `storeSecrets` or something like this.

Reply:
how about `configureSecrets`?

Reply:
I don't think "store" implies it's an update, you can store something for the first time :-)